src/interface/data_collection.cpp

```diff
@@ -62,6 +63,24 @@ DataCollection<MeshData<Real>>::GetOrAdd(int gmg_level, const std::string &mbd_l
                  GridIdentifier::two_level_composite(gmg_level))[partition_id]);
 }
 
+template <class T>
+const std::shared_ptr<T> &DataCollection<T>::Get(const std::string &name) const {
+  if constexpr (std::is_same_v<T, MeshData<Real>>) {
+    return Get(name, pmy_mesh_->GetBasePartition());
+  } else {
+    return Get(name, std::make_shared<int>());
```

Comment:
ok what is happening here?

Reply:
oh, sorry. That was a gross hack to get things working. For `Get` the second argument is only used for creating the key and for `MeshBlockData` the second argument ends up being ignored for getting the key (the key is just `name`). This was just a random thing to pass.

Reply:
We should make a comment to that effect here then. Or clean this up?

Reply:
Yeah I'd be open to splitting them. It seemed like a good idea for them to be the same when we did it...